### PR TITLE
merge to redux-saga 1.0.0, api breaking changes included.

### LINKED ISSUE #6134: Problems getting Next-redux-saga example to work with "redux-saga": "^1.0.0"
Hi -- I'm new to Next.js and a relative redux-saga newbie -- I've been given the task to connect redux-saga to Next.js for a large project.  The example code -- [https://github.com/zeit/next.js/tree/canary/examples/with-redux-saga](https://github.com/zeit/next.js/tree/canary/examples/with-redux-saga) -- works fine, but it didn't work when I moved it to my project.  Specifically storing to the store seemed to happen after all the server-side page renders or not at all.  The only difference I could notice was that with redux-saga 0.16 (I believe) that comes with the example code there is a field `done` in the saga task, whereas with redux-saga 1.0.0, which is being used in my project, there is a `toPromise` field.  I don't know if this makes the difference.  In any case it seems that my reducers are running, but I don't end up with any data in my store.  Any advice welcome!

### PR DESCRIPTION
**Pull request by bbortt**
Fixes #6134

Hi there
I noticed you have not yet included the api breaking changes in `"redux-saga": "1.0.0"`. Therefore I felt free to upgrade the dependencies in `examples/with-redux-saga`.
I do not know anything about apollo nor graphql, that is why I did not upgrade `examples/with-apollo-and-redux-saga`. But, I think you should do this on occasion.
Keep the great work up.
Regards

### CODE REVIEW

**Review by timneutkens — APPROVED** [review]
Thank you!

**Comment by timneutkens** [comment]
Could you fix the linting error?

File: examples/with-redux-saga/saga.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* global fetch */
 
-import { delay } from 'redux-saga'
+import { delay } from 'redux-saga/effects'
```
Comment (timneutkens) [inline examples/with-redux-saga/saga.js]:
```suggestion
```

File: examples/with-redux-saga/saga.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 import { all, call, put, take, takeLatest } from 'redux-saga/effects'
```
Comment (timneutkens) [inline examples/with-redux-saga/saga.js]:
```suggestion
import { all, call, put, take, takeLatest, delay } from 'redux-saga/effects'
```

Comment (bbortt) [reply examples/with-redux-saga/saga.js]: Seems like I've been a bit too tired yesterday evening. There you go: https://github.com/zeit/next.js/pull/6109/commits/6f7191dc1c2db81982b331a01943201a0abd7eb4.

**Comment by robert-barcelona** [comment]
@bbortt , @timneutkens  -- Does the new sample code work with 'next-redux-saga' 3.0.0?  I was having trouble merging the sample code into a project I'm working on and tracked it down to the changes in 'redux-saga' 1.0.0 -- specifically I believe that 'next-redux-saga'  is relying on the field 'done' in the saga task in 0.16 whereas this has been changed to 'toPromise' in 1.0.0.  Even after making the changes that @bbortt made above, I wasn't able to get the sample code to work with 1.0.0.

**Comment by bbortt** [comment]
Oh I see. Actually I'm wondering why it compiled then. Let me reply you as soon as possible.

**Comment by bbortt** [comment]
@robert-barcelona The only redux-saga import in `next-redux-saga` is
* `import {delay} from 'redux-saga'`
* *test scoped*!

Therefore, I think the upgrade to version `redux-saga: 1.0.0` may not have caused your broken build.
And just to be sure I opened another [pull-request to `next-redux-saga`](https://github.com/bmealhouse/next-redux-saga/pull/29).

**Comment by bbortt** [comment]
@robert-barcelona Yes, this may actually have broken your build.. I'm on a [pull-request to `next-redux-saga`](https://github.com/bmealhouse/next-redux-saga/pull/29). Sorry for causing you troubles!

**Pull request merged**